### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by arshiaar**
Dear Shakes and tutoring team,

Please review this pull request for my final submission for the COMP3710 Report. 

### About
This submission contains an implementation of the Improved UNET performing skin leison segmentation on the 2018 ISIC Dataset. The model architecture uses PyTorch for implementation and achieved a dice similarity coefficient of 0.86. 
The README.md provides an indepth explanation of model architecture, data and preprocessing, the model and results. 
It contains information on using this repository as well as reproducing this result. 

### Files
This submission includes the following files:
- README.md: contains documentation about the entire project 
- modules.py: Contains the building blocks of the code and ImprovedUnet architecture 
- train.py: Main driver of the project, contains training and validation code and plots losses.
- predict.py: Contains code for output visualisations and testing a loaded model from the train.py
- assets folder: contains assets used in README documentation.

### Results 
A dice coefficient of 0.86 was achieved on the test set by the Improved Unet Model after training for 50 epochs.

I look forward to your feedback. 

Many thanks, 
Arshia Sharma 
44822961

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
This is an initial inspection, no action is required at this point

- Well organized files.
- Algorithm solves the problem appropriately. Average Dice Coefficient in testing dataset: 0.86. Good output.
- Correctly using PyTorch to construct the Improved U-Net models and functions, but no data augmentation.
- Good coding design.
- Sufficient docstring and comment in the code, but need some references.
- Easy difficulty problem, no higher difficulty marks.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, no augmentation -1
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
